internal/zip/zip.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zip
+
+import (
+	"archive/zip"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/spf13/afero"
+	"io"
+	"path/filepath"
+)
+
+func Create(fs afero.Fs, zipFileName string, files []string, preservePath bool) error {
+	zipFile, err := fs.Create(zipFileName)
+	if err != nil {
+		return err
+	}
+	defer zipFile.Close()
+
+	zipWriter := zip.NewWriter(zipFile)
+	defer zipWriter.Close()
+
+	for _, f := range files {
+		err = addFileToZip(fs, zipWriter, f, preservePath)
+		if err != nil {
+			log.Warn("Unable to add %s file to support archive: %w", f, err)
```

Comment:
I can understand continuing and adding all the files we can to the archive, but shouldn't we still return an error in the end if we failed to add some of the files?